### ISSUE TITLE
Add createauxblock RPC command to 1.14.x version to faciliate minining pools' adoption

### ISSUE THREAD

**Issue by MishaBoar** [dogecoin/dogecoin#2630]
# Feature Request

User https://twitter.com/realalirez has been contacting mining pools in the past days. Several mining pools are not upgrading to newer builds because they do not include the createauxblock RPC command, which, in the past, it was discussed might have been added in Dogecoin Core 1.17 (see @rnicoll Post in 2019: https://rnicoll.name/dogecoin/1-17-mining/index.html). They could instead use getauxblock, but I think they are running some automated scripts across several AuxPow coins' pools and this would facilitate the process. 

**Describe the Feature Request**
Add createauxblock/submitnewblock RPC commands, as seen in Namecoin 1.17 to make mining pools happy.

**Describe Preferred Solution**
Grab the changes from Namecoin 1.17+ and port them into the 1.14.x builds.

**Describe Alternatives**
Ask the miners to just use the old commands!

**Related Code**
https://github.com/namecoin/namecoin-core/blob/ca8028799cd6a8dfab63600f089f0d10a9c82d41/src/rpc/mining.cpp#L1263

**Comment by patricklodder** [comment]
I think that the current share any miner has is irrelevant to the decision whether to include this or not. We knew that it had gone missing for a little over a month now, and we hadn't prioritized it because we had no indication that anyone needed it right now. Now we do, so we should act accordingly: the request is there, let's not procrastinate.

> I think the update might not be trivial 

It's relatively straightforward because we don't need to change existing logic to bring these in. The functionality can be isolated now and if we want to consolidate later, we can do that at a time that we can be more confident about the quality of the code. We can tag it "experimental".

> might require a lot of non trivial and prolonged testing

It will need a good rpc test and it needs to be compared with what btccom@46c85c5 does - either to follow that as close as possible, or maybe it should just be derived from that. Prolonged testing, always, but waiting with the release of a feature like this will not help testing at all - it will just delay testing and create barriers. My requirement was to **_depreciating_** `getauxblock`, not so much to **_re-introducing_** these methods.

TLDR; Rocket-science detectors are sitting at a hard zero. Let's do this.

**Comment by patricklodder** [comment]
> On the two quirks

About the first quirk: like I said it's a feature that I think makes sense and it doesn't hurt anyone that uses a single address, the only overhead introduced for those that don't use it is vectorization overhead, and a 3-line patch.

> I don't think it's crazy to ask them to support adding a `_` in if needed/remove if not needed.

Re-reading bullet point 1 on the v1.14.2 release notes and your work in dbfb33ab shows a bit of a different story: the underscore made it into 1.14.0 and then there were issues (I was told, also by you, that the issues were raised directly by miners) and it was reverted back without the underscore, i.e. following Dogecoin API instead of Namecoin API. Then if I add to that the fact that some miners run a separately patched/maintained Dogecoin Core just because those Namecoin RPC methods were removed, I feel that it shows that not being inclusive may lead to difficulties when you want to get something adapted - like, say, a fee policy change (to just make up a crazy example on the spot.)

If then we're going to take a "hard stand" and make an issue about diverging APIs between 2 coins, the thing that - based on the past outcome - will likely happen is that someone will just fork and change the API to what they need it to be. Kind of like what was done now, because there is no rule that says you have to run the exact version that is offered here. I personally don't do that either.

So the question then becomes: is facilitating optional compatibility with Namecoin's API worth the effort? I think the effort is low, especially when we stop doing these endless series of rewrites - which we're discussing in #2273 - and just have a stable 
base which we actually maintain. We're talking 30 lines of code and about the same quantity of qa test code, if that. If that prevents creating contention with/between miners and making their upgrades easier, then I think that's a reasonable thing to do. But that's just my take...

I guess what it comes down to is: do developers dictate or enable? Facilitate or rule? Do we draw a line around the majority and say "that's it", or do we try to make the majority as large as possible?

**Comment by patricklodder** [comment]
@michilumin Since this was starting to look stalled, I have pulled in your work from michilumin@1c5b9b3, added an extensive RPC test and fixed the issue with the address cache (retaining your commit plus some fixes). Will open PR shortly.

**Comment by patricklodder** [comment]
This is brought back now and #2654 should make it even easier for pools that were using the btccom version to migrate to 1.14.5 - I'm closing this.
